src/hotspot/share/opto/type.cpp

```diff
@@ -4036,15 +4037,24 @@ int TypeInstPtr::hash(void) const {
 //------------------------------dump2------------------------------------------
 // Dump oop Type
 #ifndef PRODUCT
-void TypeInstPtr::dump2( Dict &d, uint depth, outputStream *st ) const {
+void TypeInstPtr::dump2( Dict &d, uint depth, outputStream* st ) const {
   // Print the name of the klass.
   klass()->print_name_on(st);
 
   switch( _ptr ) {
   case Constant:
-    // TO DO: Make CI print the hex address of the underlying oop.
     if (WizardMode || Verbose) {
-      const_oop()->print_oop(st);
+      st->print(" ");
+      stringStream ss;
+      const_oop()->print_oop(&ss);
+
+      {
+        ResourceMark rm;
+        char* buf = os::strdup(ss.as_string());
+        StringUtils::tr_delete(buf, "\n");
+        st->print_raw(buf);
+        os::free(buf);
```

Comment:
There is no need to use `os::strdup` because `as_string` creates a copy.
I've looked in stringUtils.cpp and found `replace_no_expand`.  The code can be rewritten:
```
char *buf = ss.as_string();
StringUtils::replace_no_expand(buf, "\n", "");
st->print_raw(buf);
```
With this code, `tr_delete` is redundant.

Reply:
> oh, thanks for the head-up. I'm happy to remove os::strdup and os::free pair.
> it seems that replace_no_expand is cumbersome to do what tr_delete does. let me see how it works.

I don't see why it is cumbersome. IMHO, it is logically consistent: replace substring with an empty string without expanding the buffer. The main value is the amount of written code.

Reply:
oh, by means "cumbersome",  I just felt that it's easier to sweeping chars than substrings in my case.  but I has verified `replace_no_expand(buf, "\n", "")` has the same effect.  I took you advice.  less code is less chance to make mistake. 

Updated this PR. I also verified that it has the same results for `-XX:+Verbose -XX:+PrintIdeal`.